samples/java-protobuf-web-resources/src/main/java/com/example/FileServiceAction.java

```diff
@@ -47,7 +48,11 @@ private  Effect<HttpBody> loadFile(String dir, String file) {
     String fullPath = baseDir+dir+"/"+file;
     try {
       // tag::200-ok[]
-      byte[] byteArray = getClass().getResourceAsStream(fullPath).readAllBytes();
+      InputStream inputStream = getClass().getResourceAsStream(fullPath);
+      if(null == inputStream) {
+        throw new NoSuchFileException("File " + fullPath + " not found");
+      }
```

Comment:
Should really be a return with a not-found grpc status code (which should lead to a 404 response from the proxy)  rather than an exception leading to 500 internal.

Reply:
Maybe a `throw` should not be used here, but right now this one gets caught below and turned into a `effects().ignore();` which gets a 404 in the browser, which I was hoping to express that the requested file does not exist.

Reply:
Ah, I missed the end of the method with the specific catch.